text/0003-media-global-cluster.md

```diff
@@ -0,0 +1,101 @@
+- Feature Name: media-global-cluster
+- Start Date: 2024-02-02
+- RFC PR: [8xff/rfcs#0003](https://github.com/8xff/rfcs/pull/0003)
+
+# Summary
+[summary]: #summary
+
+This RFC proposes way to orchestrate cluster topology and how we select the best node for each request.
+
+# Motivation
+[motivation]: #motivation
+
+In live stream application, the most important thing is the quality of the stream. We need to ensure that the stream is delivered to the client with the best quality. To do that, we need to select the best node to serve each request. The best node is the node which is closest to the client and have the best network connection to the source.
+
+# User Benefit
+
+The user can setup a cluster across multiple regions and the system will automatically select the best node to serve each request. Mostly steps is automated and the user only need to setup the cluster and the system will do the rest.
+
+# Design Proposal
+
+## Abstract Design
+
+### Gateway
+
+![Single zone](./0003-media-global-cluster/single-zone.excalidraw.png)
+
+We propose a gateway node, which will take care to route each request to best node. The process is described as below:
+
+Update info process:
+
+- Each node will broadcast its information to all gateway nodes in same zone.
```

Comment:
For simple, each node should send ping to only one gateway node, the gateway node will do the rest